plugin-flex-ts-template-v2/src/feature-library/contact-history/custom-components/ContactHistoryView.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import { templates, Template } from '@twilio/flex-ui';
+import { useDispatch, useSelector } from 'react-redux';
+import { Button } from '@twilio-paste/core/button';
+import { Flex } from '@twilio-paste/core/flex';
+import { Box } from '@twilio-paste/core/box';
+import { Table, THead, TBody, Tr, Th } from '@twilio-paste/core/table';
+import { AlertDialog } from '@twilio-paste/core/alert-dialog';
+
+import { clearContactList } from '../flex-hooks/state';
+import RecentContacts from '../utils/ContactsUtil';
+import ContactRecord from './ContactRecord';
+import { Contact } from '../types';
+import AppState from '../../../types/manager/AppState';
+import { reduxNamespace } from '../../../utils/state';
+import { StringTemplates } from '../flex-hooks/strings';
+
+const ContactHistory = () => {
+  const [confirmClearHistory, setConfirmClearHistory] = useState(false);
+
+  const contactData = useSelector((state: AppState) => {
+    return { contactList: state[reduxNamespace]?.contactHistory?.contactList };
+  });
+
+  const contactList = contactData?.contactList || [];
+  const dispatch = useDispatch();
+
+  const clearHistory = () => {
+    setConfirmClearHistory(false);
+    dispatch(clearContactList());
+    RecentContacts.clearContactList();
+  };
+
+  const closeClearHistory = () => setConfirmClearHistory(false);
+
+  return (
+    <Flex width="100%">
+      <Flex vertical width="100%" grow shrink>
+        <Box padding="space40">
+          <Button variant="primary" onClick={async () => setConfirmClearHistory(true)}>
+            <Template source={templates[StringTemplates.ClearHistory]} />
+          </Button>
+          <AlertDialog
+            heading={templates[StringTemplates.ClearHistory]()}
+            isOpen={confirmClearHistory}
+            onConfirm={async () => clearHistory()}
+            onConfirmLabel={templates.ConfirmableDialogConfirmButton()}
+            onDismiss={closeClearHistory}
+            onDismissLabel={templates.ConfirmableDialogCancelButton()}
+          >
+            <Template source={templates[StringTemplates.ClearHistoryDialog]} />
+          </AlertDialog>
+        </Box>
+        <Box overflowY="auto" width="100%" padding="space40">
+          <Table variant="borderless">
```

Comment:
Could we add an empty state to the table? It would be nice if it had a placeholder like "Calls and chats will appear here once they are completed."

Reply:
Done